### DISCUSSION TITLE
Adding custom completions to auto-suggest (without TAB)

### DISCUSSION THREAD

**Discussion by amitkot** [xonsh/xonsh#5219] [Q&A]
I want to extend Xonsh by allowing a custom auto-suggest text that will be suggested while typing a command.
The current completion hooks fit my needs but can only be activated when the user presses the `TAB` key, while my need is for suggesting automatically completions while typing without requiring the `TAB` key press.

Is it possible to hook into the auto-suggest flow this way?

**Comment by anki-code** [comment]
xonsh is using [prompt-toolkit](https://github.com/prompt-toolkit/python-prompt-toolkit) for prompt features. Try to dive into it.
